crypto/multisig/src/lib.rs

```diff
@@ -445,6 +502,148 @@ mod test {
+        let org2_signer3 = Ed25519Pair::from_random(&mut rng);
+
+        let individual_signer = Ed25519Pair::from_random(&mut rng);
+
+        // Org 1 requires 2-out-of-2
+        let org1_signer_set = SignerSet::new(
+            vec![org1_signer1.public_key(), org1_signer2.public_key()],
+            vec![],
+            2,
+        );
+
+        // Org 2 requires 2-out-of-3
+        let org2_signer_set = SignerSet::new(
+            vec![
+                org2_signer1.public_key(),
+                org2_signer2.public_key(),
+                org2_signer3.public_key(),
+            ],
+            vec![],
+            2,
+        );
+
+        // Test a signer set that requires the two orgs and an individual signer.
+        let signer_set = SignerSet::new(
+            vec![individual_signer.public_key()],
+            vec![org1_signer_set.clone(), org2_signer_set.clone()],
+            3,
+        );
+
+        let multi_sig = MultiSig::new(vec![
+            // Org 1 satisfied
+            org1_signer1.try_sign(message.as_ref()).unwrap(),
+            org1_signer2.try_sign(message.as_ref()).unwrap(),
+            // Org 2 unsatisfied
+            org2_signer1.try_sign(message.as_ref()).unwrap(),
+            // Individual signer satisfied
+            individual_signer.try_sign(message.as_ref()).unwrap(),
+        ]);
+        assert!(signer_set.verify(message.as_ref(), &multi_sig).is_err());
+
+        let multi_sig = MultiSig::new(vec![
+            // Org 1 satisfied
+            org1_signer1.try_sign(message.as_ref()).unwrap(),
+            org1_signer2.try_sign(message.as_ref()).unwrap(),
+            // Org 2 satisfied
+            org2_signer1.try_sign(message.as_ref()).unwrap(),
+            org2_signer2.try_sign(message.as_ref()).unwrap(),
+        ]);
+        assert!(signer_set.verify(message.as_ref(), &multi_sig).is_err());
+
+        let multi_sig = MultiSig::new(vec![
+            // Org 1 satisfied
+            org1_signer1.try_sign(message.as_ref()).unwrap(),
+            org1_signer2.try_sign(message.as_ref()).unwrap(),
+            // Org 2 satisfied (and has an extra signature, but this still doesn't count towards
+            // the top level threshold)
+            org2_signer1.try_sign(message.as_ref()).unwrap(),
+            org2_signer2.try_sign(message.as_ref()).unwrap(),
+            org2_signer3.try_sign(message.as_ref()).unwrap(),
+        ]);
+        assert!(signer_set.verify(message.as_ref(), &multi_sig).is_err());
+
+        // Providing the same signers multiple times does not change the result.
+        let multi_sig = MultiSig::new(vec![
+            // Org 1 satisfied
+            org1_signer1.try_sign(message.as_ref()).unwrap(),
+            org1_signer2.try_sign(message.as_ref()).unwrap(),
+            org1_signer2.try_sign(message.as_ref()).unwrap(),
+            org1_signer2.try_sign(message.as_ref()).unwrap(),
+            // Org 2 not satisfied
```

Comment:
:wrench: I think org2 is satisfied here, it's a 2of 3 and there are 2.
```suggestion
            // Org 2 satisfied
```